zds/tutorialv2/views/statistics.py

```diff
@@ -85,23 +85,30 @@ def get_all_statistics(self, urls, start, end, methods):
             response_matomo = requests.post(url=self.matomo_api_url, data=data_request)
             data = response_matomo.json()
             if isinstance(data, dict) and data.get("result", "") == "error":
-                data = {}
                 self.logger.error(data.get("message", "Something failed with Matomo reporting system."))
                 messages.error(
-                    self.request, data.get("message", _(f"Impossible de récupérer les statistiques du site."))
+                    self.request, data.get("message", _("Impossible de récupérer les statistiques du site."))
                 )
-
-            for index, method_url in enumerate(itertools.product(methods, urls)):
-                method = method_url[0]
-                data_structured[method].append(data[index])
-
-            return data_structured
+                return {}
+            else:
+                for index, method_url in enumerate(itertools.product(methods, urls)):
+                    if isinstance(data[index], dict) and data[index].get("result", "") == "error":
+                        self.logger.error(data[index].get("message", "Something failed with Matomo reporting system."))
+                        messages.error(
+                            self.request,
+                            data[index].get("message", _("Impossible de récupérer les statistiques du site.")),
+                        )
+                        return {}
+
+                    method = method_url[0]
+                    data_structured[method].append(data[index])
+
+                return data_structured
         except Exception:
-            data = {}
             self.logger.exception(f"Something failed with Matomo reporting system.")
```

Comment:
Ici tu peux enlever le `f` ici aussi.

```suggestion
            self.logger.exception("Something failed with Matomo reporting system.")
```